doc/guides/tfm/overview.rst

```diff
@@ -0,0 +1,266 @@
+
+Key and secret management is a critical part of any secure device. You need to
+ensure that key material is available to regions that require it, but not to
+anything else, and that it is stored securely in a way that makes it difficult
+to tamper with or maliciously access.
+
+The **Internal Trusted Storage** service in TF-M is used by the **PSA Crypto**
+service (which itself makes use of mbedtls) to store keys, and ensure that
+private keys are only ever accessible to the secure processing environment.
+Crypto operations that make use of key material, such as when signing payloads
+or when decrypting sensitive data, all take place via key handles. At no point
+should the key material ever be exposed to the NS environment.
+
+One exception is that private keys can be provisioned into the secure
+processing environment as a one-way operation, such as during a factory
+provisioning process, but even this should be avoided where possible, and a
+request should be made to the SPE (via the PSA Crypto service) to generate a
+new private key itself, and the public key for that can be requested during
+provisioning and logged in the factory. This ensures the private key
+material is never exposed, or even known during the provisioning phase.
+
+TF-M also makes extensive use of the **Hardware Unique Key (HUK)**, which
+every TF-M device must provide. This device-unique key is used by the
+**Protected Storage** service, for example, to encrypt information stored in
+external memory, such that removing the memory and placing it on a new device,
+the contents of flash memory can't be decrypted since each device has it's own
+unique HUK used while encrypted the memory contents the first time.
+
+HUKs provide an additional advantage for developers, in that they can be used
+to derive new keys, and the **derived keys** don't need to be stored since
+they can be regenerated from the HUK at startup, using an additional salt/seed
+value (depending on the key derivation algorithm used). This removes the
+storage issue and a frequent attack vector. The HUK itself it usually highly
+protected in secure devices, and inaccessible directly by users.
+
+``TFM_CRYPTO_ALG_HUK_DERIVATION`` identifies the default key derivation
+algorithm used if a software implementation is used. The current default
+algorithm is ``HKDF`` (RFC 5869) with a SHA-256 hash. Other hardware
+implementations may be available on some platforms.
+
+Non-Secure Processing Environment
+=================================
+
+Zephyr is used for the NSPE, using a board that is supported by TF-M where the
+:kconfig:`CONFIG_BUILD_WITH_TFM` flag has been enabled.
+
+Generally, you simply need to select the ``*_ns`` variant of a valid target
+(for example ``mps2_an521_ns``), which will configure your Zephyr application
+to run in the NSPE, correctly build and link it with the TF-M secure images,
+sign the secure and non-secure images, and merge the three binaries into a
+single ``tfm_merged.hex`` file.
```

Comment:
```suggestion
single ``tfm_merged.hex`` file. The :ref:`west flash <west-flashing>` command
will flash ``tfm_merged.hex`` by default in this configuration.
```

Knowing this would have saved me a few hairs the first time I was trying to use nrf5340dk_nrf5340_cpuapp_ns.